src/main/java/io/quarkus/gizmo/ClassTransformer.java

```diff
@@ -0,0 +1,196 @@
+package io.quarkus.gizmo;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+
+/**
+ * Allows transforming an existing class by wrapping an ASM {@link ClassVisitor}.
+ * The {@link #applyTo(ClassVisitor)} method must be called before the wrapped
+ * {@code ClassVisitor} is used in any way and the result should be used instead
+ * of the original visitor.
+ * <p>
+ * In other words, this class acts like a builder. First, the class transformation
+ * must be described using {@code addMethod()}, {@code addField()}, etc. Then,
+ * a {@code ClassVisitor} must be created using {@code applyTo()}. Afterwards,
+ * the {@code ClassTransformer} instance should be discarded.
+ */
+public class ClassTransformer {
+
+    private static class NamedDescriptor {
```

Comment:
I would move this class down in the source file... it's private, not part of the API, so I see no reason to show it as the first thing when the `ClassTransformer` source is opened.

Reply:
Fair enough, moving it to the end should be fine.

Reply:
Done.